### DISCUSSION TITLE
Detect and Report Errors for Duplicate Names Between Files and Folders

### DISCUSSION THREAD

**Discussion by babakfp** [withastro/roadmap#842] [Proposal]
### Body

# Summary

In this sort of structure:

📁 `src/content/docs/guides/`
📄 `src/content/docs/guides.md`

I expect there to be an error, telling the user to move `guides.md` into `guides/` and rename it to `index.md`.

# Goals

- Clean structure
- Organized structure

**Comment by florian-lefebvre** [comment]
I don't see why that should error. Both are valid and come down to personal preferences

**Reply by alexanderniebuhr** [reply]
I guess the idea is to warn/error, if the user has both at the same time in their project, since those would collide.

However, I'm unsure if that should be a framework task to warn the user about. Does anyone else has such warnings?

**Reply by florian-lefebvre** [reply]
Ah I misread the proposal then! I think collision is already handled tho (https://stackblitz.com/edit/github-jekaf8?file=src%2Fcontent%2Fblog%2Ffirst-post%2Findex.md,src%2Fcontent%2Fblog%2Ffirst-post.md):

```
12:23:03 [ERROR] [DuplicateContentEntrySlugError] blog contains multiple entries with the same slug: `first-post`. Slugs must be unique.

Entries: 
- /src/content/blog/first-post.md
- /src/content/blog/first-post/index.md
  Error reference:
    https://docs.astro.build/en/reference/errors/duplicate-content-entry-slug-error/
  Stack trace:
    at new AstroError (/home/projects/github-jekaf8/node_modules/astro/dist/core/errors/errors.js:37:5)
```

Or am I still misunderstanding?
